Packs/MicrosoftAdvancedThreatAnalytics/Integrations/MicrosoftAdvancedThreatAnalytics/MicrosoftAdvancedThreatAnalytics.py

```diff
@@ -0,0 +1,348 @@
+        min_severity: int,
+        activity_type_to_fetch: List
+) -> Tuple[Dict[str, str], List[Dict[str, str]]]:
+    last_fetch = last_run.get('last_fetch', '') if last_run.get('last_fetch') else first_fetch_time
+    last_fetch_dt = parse(last_fetch).replace(tzinfo=utc)
+    latest_start_time = parse(last_fetch).replace(tzinfo=utc)
+
+    incidents_fetched = 0
+    incidents: List[Dict[str, Any]] = []
+
+    suspicious_activities = client.get_suspicious_activity_request()
+    suspicious_activities_list = suspicious_activities if isinstance(suspicious_activities, list) \
+        else [suspicious_activities]
+    demisto.debug(suspicious_activities_list)
+    for activity in suspicious_activities_list:
+        if incidents_fetched == max_results:
+            break
+        activity_id = activity.get('Id', '')
+        activity_status = activity.get('Status', '')
+        activity_type = activity.get('Type', '')
+        activity_severity = activity.get('Severity', '')
+        if activity_status_to_fetch and activity_status not in activity_status_to_fetch:
+            demisto.debug(f'Skipping suspicious activity {activity_id} with status {activity_status}')
+            continue
+        if activity_type_to_fetch and activity_type not in activity_type_to_fetch:
+            demisto.debug(f'Skipping suspicious activity {activity_id} with type {activity_type}')
+            continue
+        if SEVERITY_TRANSLATION[activity_severity] < min_severity:
+            demisto.debug(f'Skipping suspicious activity {activity_id} with severity {activity_severity}')
+            continue
+        activity_start_time = activity.get('StartTime', '')
+        activity_start_time_dt = parse(activity_start_time).replace(tzinfo=utc)
+        if activity_start_time_dt > latest_start_time:
+            incidents.append({
+                'name': f'{activity_type} - {activity_id}',
+                'occurred': activity_start_time,
+                'rawJSON': json.dumps(activity)
+            })
+            if activity_start_time_dt > last_fetch_dt:
+                last_fetch_dt = activity_start_time_dt
+                last_fetch = activity_start_time
+            incidents_fetched += 1
+    next_run = {'last_fetch': last_fetch}
+    return next_run, incidents
+
+
+def main() -> None:
+    params = demisto.params()
+    base_url = urljoin(params['url'], '/api/management')
+    username = params['credentials']['identifier']
+    password = params['credentials']['password']
+    verify_certificate = not params.get('insecure', False)
+    proxy = params.get('proxy', False)
+
+    demisto.debug(f'Command being called is {demisto.command()}')
+    try:
+        client = Client(
+            base_url=base_url,
+            auth=HttpNtlmAuth(username, password),
+            verify=verify_certificate,
+            proxy=proxy
+        )
+
+        if demisto.command() == 'test-module':
+            result = test_module(client)
```

Comment:
didn't we agree that the `test-module` should also run the `fetch-incidents` flow when enabled?